### PR TITLE
[Core][KratosComponents] add proper errors & tests

### PR DESCRIPTION
**Pull request by philbucher**
This PR adds proper error messages where previously it would segfault

Tests are added

### CODE REVIEW

**Review by pooyan-dadvand — COMMENTED** [review]
Very important addition! Thanks!! 

Are you sure that we should not debug only check in Get functions?

**Comment by philbucher** [comment]
> Are you sure that we should not debug only check in Get functions?

I don't have a strong opinion here, I will change it do debug only for the Get functions

For the Add-functions I will leave it because those typically only get called at the beginning of an execution when the Apps are being loaded.

**Comment by philbucher** [comment]
@pooyan-dadvand done, now the `Get`-functions only throw in Debug

Test is adapted

**Comment by philbucher** [comment]
I used `KRATOS_DEBUG` to implement the fastest version possible

**Comment by philbucher** [comment]
wait, some more updates coming

File: kratos/includes/kratos_components.h [hunk 1]
```diff
@@ -80,23 +81,19 @@ class KRATOS_API(KRATOS_CORE) KratosComponents
...
-        }
 
-        std::stringstream err_msg;
-        err_msg << "The component \"" << Name << "\" is not registered!\n"
-                << "Maybe you need to import the application where it is defined?\n"
-                << "The following components of this type are registered:" << std::endl;
-        KratosComponents instance; // creating an instance for using "PrintData"
-        instance.PrintData(err_msg);
-        KRATOS_ERROR << err_msg.str() << std::endl;
+        KRATOS_DEBUG_ERROR_IF(it_comp == msComponents.end()) << GetMessageUnregisteredComponent(Name) << std::endl;
```
Comment (philbucher) [inline kratos/includes/kratos_components.h:94]: @pooyan-dadvand now it works completely without `KRATOS_DEBUG`, also I think it is quite easy to read/understand, please have a lool

**Review by pooyan-dadvand — APPROVED** [review]
Thanks! :+1:

**Pull request merged**